### PR TITLE
Add Bagit validation (#93)

### PR DESCRIPTION
**Pull request by mcantelon**
Added validation of Bag before it's unbagged.

### CODE REVIEW

File: internal/workflow/preprocessing_test.go [hunk 1]
```diff
@@ -135,6 +137,14 @@ func (s *PreprocessingTestSuite) SetupTest(cfg *config.Configuration) {
 	s.sipPath = sp
 
 	// Register activities.
+	validator, err := bagit_gython.NewBagIt()
+	if err != nil {
+		s.T().Fatalf("initialize bagit: %v", err)
+	}
+	s.env.RegisterActivityWithOptions(
+		bagvalidate.New(validator).Execute,
```
Comment (djjuhasz) [inline internal/workflow/preprocessing_test.go:145]: You can just use a nil validator here (e.g. `bagvalidate.New(nil).Execute`) and it won't make any difference to the results.  All of the activity calls in these workflow tests are mocked, so the activity is never actually run and the `validator.Validate()` method is never called.

**Review by djjuhasz — CHANGES_REQUESTED** [review]
Looks good @mcantelon, just one change request.

**Comment by mcantelon** [comment]
I've made that change, @djjuhasz ! Let me know if anything else needs tweaking!

File: internal/workflow/preprocessing_test.go (hunk 1 above)
Comment (djjuhasz) [reply internal/workflow/preprocessing_test.go:142]: @mcantelon you don't even need to use the `bagvalidate.NewNoopValidator()` I've discovered. You can literally just pass a `nil` for the validator:

```go
bagvalidate.New(nil).Execute,
```

**Review by djjuhasz — CHANGES_REQUESTED** [review]

**Comment by mcantelon** [comment]
Ah sweet! Fixed! Thanks @djjuhasz !

**Review by djjuhasz — APPROVED** [review]
Looking good! :shipit:

**Pull request merged**